### ISSUE TITLE
Custom dataset is using CPU instead GPU on classification

### ISSUE THREAD

**Issue by fhilipecrash** [ultralytics/ultralytics#21724]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

## System Informations

- OS: Ubuntu 24.04.3 LTS
- CPU: 12th Gen Intel® Core™ i5-12400 × 12
- GPU: NVIDIA GeForce GTX 1660 SUPER
- RAM: 8,0 GiB
- NVIDIA Driver Version: 575.64.03
- CUDA Version: 12.0.140
- Python Version: 3.12.3
- YOLO Version: 11
- Ultralytcs Version: 8.3.160
- PyTorch Version: 2.7.1+cu126

## The question

I noticed that the classification training was very slow, and when I checked the system resource usage, I noticed that YOLO was using much more CPU than GPU. So I used the datasets recommended in the documentation to see if this was the case, and I saw that they were using the GPU normally, as shown in the images.

### YOLO classification training running with my custom dataset
<img width="1163" height="794" alt="Image" src="https://github.com/user-attachments/assets/386c98e1-0b3b-473f-9430-e10458042702" />

### YOLO classification training running with cifar100 dataset
<img width="1163" height="794" alt="Image" src="https://github.com/user-attachments/assets/40379f74-8542-4c71-ae1e-f3d111cae0b5" />

That's strange because both training use CUDA and create python GPU process visible in `nvidia-smi` but my dataset somehow don't use GPU effectively.

### YOLO process with my dataset

<img width="1163" height="794" alt="Image" src="https://github.com/user-attachments/assets/4a74bae5-114d-47f7-8a29-4db7e37683a1" />

### YOLO process with cifar100 dataset

<img width="1163" height="794" alt="Image" src="https://github.com/user-attachments/assets/52d5dada-e6ea-4993-befd-9bb337a74ead" />

I've use this command line to train both only changing `data` param to `dcm` that's name of my dataset
```bash
poetry run yolo classify train model=yolo11n-cls.pt data=cifar100 epochs=100 imgsz=224
```

My dataset directory have this structure, i've mounted this based in docs.

```bash
dcm/
|
|-- train/
|   |-- b1/
|   |   |-- 001_b1.png
|   |   |-- 002_b1.png
|   |   |-- ...
|   |
|   |-- others/
|   |   |-- 001_others.png
|   |   |-- 002_others.png
|   |   |-- ...
|
|-- test/
|   |-- b1/
|   |   |-- 001_b1.png
|   |   |-- 002_b1.png
|   |   |-- ...
|   |
|   |-- others/
|   |   |-- 001_others.png
|   |   |-- 002_others.png
|   |   |-- ...
|   |
```

I really don't know what i'm doing wrong to my dataset not use GPU aceleration to train model, please help me.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @fhilipecrash, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will review your issue and assist you soon.

We recommend a visit to the Docs for quick guidance where you can find many Python and CLI usage examples and where many of the most common questions may already be answered:
- Docs: https://docs.ultralytics.com/
- Python: https://docs.ultralytics.com/usage/python/
- CLI: https://docs.ultralytics.com/usage/cli/

If this is a 🐛 Bug Report, please provide a minimum reproducible example to help us debug it:
- MRE guide: https://docs.ultralytics.com/help/minimum-reproducible-example/

If this is a custom training ❓ Question, please provide as much information as possible, including:
- Exact command(s) used
- Full console logs with warnings/errors
- A few example images per class
- Dataset structure and any custom transforms
- Hardware details and batch size/num_workers settings
And verify you are following our Tips for Best Training Results:
- https://docs.ultralytics.com/guides/model-training-tips/

Join the Ultralytics community where it suits you best. For real-time chat, head to Discord 🎧. Prefer in-depth discussions? Check out Discourse. Or dive into threads on our Subreddit to share knowledge with the community:
- Discord: https://discord.com/invite/ultralytics
- Discourse: https://community.ultralytics.com/
- Subreddit: https://www.reddit.com/r/Ultralytics/

## Upgrade

Please upgrade to the latest ultralytics package including all requirements in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

Requirements: https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See GCP Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/
- Amazon Deep Learning AMI. See AWS Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/
- Docker Image. See Docker Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/ <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all Ultralytics CI tests are currently passing. CI tests verify correct operation of all YOLO Modes and Tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help us investigate your report about CPU vs GPU usage during classification on your custom dataset, please add:
1. A minimal reproducible example (small subset of your dataset that reproduces the behavior) 📦
2. The exact command you run (you shared one — include any variations, batch size, workers, device flags)
3. The full terminal output from start to first few epochs
4. A sample of your dataset folder (5–10 images per class) with the same structure you described
5. Any custom augmentations/transforms or environment specifics (e.g., running via Poetry)
6. Confirmation you’ve tested after upgrading ultralytics and PyTorch per instructions above

Once we have this info, we can replicate and advise next steps. Thank you! 🙏

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

Can you also post the training logs from both runs? From the start until the first epoch.

**Comment by fhilipecrash** [comment]
> Can you provide the output after running this command in terminal: `yolo checks`?
> 
> Can you also post the training logs from both runs? From the start until the first epoch.

@Y-T-G Here's the `yolo check` output

```bash
Ultralytics 8.3.160 🚀 Python-3.12.3 torch-2.7.1+cu126 CUDA:0 (NVIDIA GeForce GTX 1660 SUPER, 5740MiB)
Setup complete ✅ (12 CPUs, 7.5 GB RAM, 286.0/287.3 GB disk)

OS                  Linux-6.8.0-71-generic-x86_64-with-glibc2.39
Environment         Linux
Python              3.12.3
Install             pip
Path                /home/fhilipe/.cache/pypoetry/virtualenvs/yolo-zkTfaz2S-py3.12/lib/python3.12/site-packages/ultralytics
RAM                 7.48 GB
Disk                286.0/287.3 GB
CPU                 12th Gen Intel Core(TM) i5-12400
CPU count           12
GPU                 NVIDIA GeForce GTX 1660 SUPER, 5740MiB
GPU count           1
CUDA                12.6

numpy               ✅ 2.3.1>=1.23.0
matplotlib          ✅ 3.10.3>=3.3.0
opencv-python       ✅ 4.11.0.86>=4.6.0
pillow              ✅ 11.2.1>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.4>=2.23.0
scipy               ✅ 1.16.0>=1.4.1
torch               ✅ 2.7.1>=1.8.0
torch               ✅ 2.7.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.22.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 7.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.3.0>=1.1.4
ultralytics-thop    ✅ 2.0.14>=2.0.0
```

And here's is the output of training with both datasets

[cifar100.txt](https://github.com/user-attachments/files/21836726/cifar100.txt)

[custom_dataset.txt](https://github.com/user-attachments/files/21836741/custom_dataset.txt)

**Comment by Y-T-G** [comment]
From the logs, both of them appear to be using GPU. The high CPU usage could be due to preprocessing. What are the resolution of the images in your dataset?

**Comment by fhilipecrash** [comment]
I checked just now that you mentioned it, and my images were in VERY large resolutions, like 1926 × 2332, while the recommended resolution would be something like 32x32 or 64x64, right? That's what I observed in the datasets shown in the docs. I'll resize the images and try training again. Thanks for showing me a way.

**Comment by Y-T-G** [comment]
The recommended would be close to what you're training at (`imgsz`). Like 256x256 for imgsz=224. You can preserve the aspect ratio. Just make the shortest side closer to 224.
